File: scenarios/approval-gates/python/src/tools.py
"""Tool functions for the approval-gates scenario.

  - get_balance   — safe read-only query (ALLOWED immediately)
  - transfer_funds — risky write operation (requires APPROVAL before execution)
"""
from __future__ import annotations

_BALANCES: dict[str, float] = {
    "acc-001": 12450.00,
    "acc-002": 3200.00,
    "acc-003": 875.50,
}


def get_balance(account_id: str) -> str:
    """Return the account balance (safe, read-only)."""
    balance = _BALANCES.get(account_id, 0.00)
    return f"${balance:,.2f}"


def transfer_funds(from_account: str, to_account: str, amount: float) -> str:
    """Transfer funds between accounts (requires approval before execution)."""
    return f"Transferred ${amount:,.2f} from {from_account} to {to_account}"

File: scenarios/approval-gates/node/src/tools.ts
export interface ToolResult {
  output: string;
}

const BALANCES: Record<string, number> = {
  "acc-001": 12450,
  "acc-002": 3200,
  "acc-003": 875.5,
};

export function getBalance(accountId: string): ToolResult {
  const balance = BALANCES[accountId] ?? 0;
  return { output: `$${balance.toLocaleString("en-US", { minimumFractionDigits: 2 })}` };
}

export function transferFunds(fromAccount: string, toAccount: string, amount: number): ToolResult {
  return { output: `Transferred $${amount.toLocaleString("en-US", { minimumFractionDigits: 2 })} from ${fromAccount} to ${toAccount}` };
}
